### DISCUSSION TITLE
API request, /proxy/network/api/,  retrun 403, Forbidden

### DISCUSSION THREAD

**Discussion by kyle-kwon** [Art-of-WiFi/UniFi-API-client#219] [Q&A]
After upgrading Unifi OS from version 3.1.14 to 3.2.12 on my UCK G2 Plus, I encountered a 403 Forbidden error when making API requests with a cookie. This issue is evident in the example below. 
```
#!/bin/bash
username=AAAA
password=BBBB
endpoint="https://127.0.0.1:443"
cookie="api-cookie.txt"

curl -ks -c "${cookie}" --cookie-jar ${cookie} -d '{"username":"'${username}'", "password":"'${password}'"}' -H "Content-Type: application/json" -X POST "${endpoint}/api/auth/login"
curl -ks -b "${cookie}" --cookie-jar ${cookie} -d '{"cmd": "speedtest-status"}' "${endpoint}/proxy/network/api/s/default/cmd/devmgr"

```

```
HTTP/2 403 
server: nginx
date: Mon, 18 Mar 2024 20:17:51 GMT
content-type: application/json
content-length: 44
referrer-policy: no-referrer
strict-transport-security: max-age=15552000; includeSubDomains
x-content-type-options: nosniff
x-dns-prefetch-control: off
x-frame-options: SAMEORIGIN
x-xss-protection: 1; mode=block

{"error":{"code":403,"message":"Forbidden"}}
```

 I'm curious if the cookie method is no longer functional. If it is indeed deprecated, could you provide any documentation on alternative methods for sending API requests?

**Comment by malle-pietje** [comment]
Why not use this PHP API client/wrapper? It takes away all of this tinkering😉
Just my 2 cents…

**Reply by kyle-kwon** [reply]
thanks, figure it out the issue. Mainly, Token key name has been changed on the new firmware.

**Reply by malle-pietje** [reply]
Thanks for the update. Can you share the details of your findings with the community?
